### PR TITLE
Add resource attributes

### PR DESCRIPTION
**Pull request by shashank11p**
Update telemetry resource attributes and add hypertrace attributes.

### CODE REVIEW

File: otel-extensions/src/main/java/org/hypertrace/agent/otel/extensions/HypertraceResourceProvider.java [hunk 1]
```diff
@@ -40,6 +51,20 @@ public Resource createResource(ConfigProperties config) {
       builder.put(ResourceAttributes.CONTAINER_ID, containerId);
     }
     builder.put(ResourceAttributes.SERVICE_NAME, agentConfig.getServiceName().getValue());
+    builder.put(ResourceAttributes.TELEMETRY_SDK_NAME, HYPERTRACE);
+    builder.put(ResourceAttributes.TELEMETRY_SDK_LANGUAGE, JAVA);
+    String agentVersion = "";
+    try {
```
Comment (pavolloffay) [inline otel-extensions/src/main/java/org/hypertrace/agent/otel/extensions/HypertraceResourceProvider.java:57]: please move this to a separate method

File: otel-extensions/src/main/java/org/hypertrace/agent/otel/extensions/HypertraceResourceProvider.java [hunk 2]
```diff
@@ -25,10 +26,20 @@
...
+import org.slf4j.LoggerFactory;
 
 @AutoService(ResourceProvider.class)
 public class HypertraceResourceProvider implements ResourceProvider {
 
+  private static final Logger log =
+      LoggerFactory.getLogger(HypertraceResourceProvider.class.getName());
+
+  private static final String HYPERTRACE = "hypertrace";
+  private static final String JAVA = "java";
```
Comment (pavolloffay) [inline otel-extensions/src/main/java/org/hypertrace/agent/otel/extensions/HypertraceResourceProvider.java:39]: this constant seems redundant

File: otel-extensions/src/main/java/org/hypertrace/agent/otel/extensions/HypertraceResourceProvider.java [hunk 3]
```diff
@@ -40,6 +51,20 @@ public Resource createResource(ConfigProperties config) {
...
+    builder.put(ResourceAttributes.TELEMETRY_SDK_LANGUAGE, JAVA);
+    String agentVersion = "";
+    try {
+      Class<?> hypertraceAgentClass =
+          Class.forName("org.hypertrace.agent.instrument.HypertraceAgent", true, null);
+      agentVersion = hypertraceAgentClass.getPackage().getImplementationVersion();
+    } catch (ClassNotFoundException e) {
+      log.warn("Could not load HypertraceAgent class");
+    }
+    builder.put(ResourceAttributes.TELEMETRY_SDK_VERSION, agentVersion);
```
Comment (pavolloffay) [inline otel-extensions/src/main/java/org/hypertrace/agent/otel/extensions/HypertraceResourceProvider.java:64]: is this correct? The SDK version does not match the agent version

File: otel-extensions/src/main/java/org/hypertrace/agent/otel/extensions/HypertraceResourceProvider.java [hunk 4]
```diff
@@ -40,6 +51,20 @@ public Resource createResource(ConfigProperties config) {
...
+    builder.put(ResourceAttributes.TELEMETRY_AUTO_VERSION, agentVersion);
```
Comment (pavolloffay) [inline otel-extensions/src/main/java/org/hypertrace/agent/otel/extensions/HypertraceResourceProvider.java:65]: can you point me to the spec that defines what version should be used here?

File: otel-extensions/src/main/java/org/hypertrace/agent/otel/extensions/HypertraceResourceProvider.java [hunk 5]
```diff
@@ -40,6 +49,25 @@ public Resource createResource(ConfigProperties config) {
       builder.put(ResourceAttributes.CONTAINER_ID, containerId);
     }
     builder.put(ResourceAttributes.SERVICE_NAME, agentConfig.getServiceName().getValue());
+    builder.put(ResourceAttributes.TELEMETRY_SDK_NAME, "hypertrace");
+    builder.put(ResourceAttributes.TELEMETRY_SDK_LANGUAGE, "java");
+    String agentVersion = getAgentVersion();
+    builder.put(ResourceAttributes.TELEMETRY_SDK_VERSION, agentVersion);
+    builder.put(ResourceAttributes.TELEMETRY_AUTO_VERSION, agentVersion);
+    builder.put(AttributeKey.stringKey(HYPERTRACE_MODULE_NAME), "java");
+    builder.put(AttributeKey.stringKey(HYPERTRACE_MODULE_VERSION), agentVersion);
```
Comment (pavolloffay) [inline otel-extensions/src/main/java/org/hypertrace/agent/otel/extensions/HypertraceResourceProvider.java:58]: @davexroth related to the wiki doc - it seems there is quite large redundancy in the version attributes. 

Could we use only a single one if the values are repeated?

Comment (davexroth) [reply otel-extensions/src/main/java/org/hypertrace/agent/otel/extensions/HypertraceResourceProvider.java:58]: I replied to your comment on the wiki.

**Review by pavolloffay — APPROVED** [review]

**Pull request merged**